processing/src/main/java/org/apache/druid/query/aggregation/any/DoubleAnyAggregatorFactory.java

```diff
@@ -0,0 +1,200 @@
+
+public class DoubleAnyAggregatorFactory extends NullableNumericAggregatorFactory<ColumnValueSelector>
+{
+  private static final Comparator<Number> VALUE_COMPARATOR = Comparator.nullsFirst(
+      Comparator.comparingDouble(Number::doubleValue)
+  );
+
+  private final String fieldName;
+  private final String name;
+  private final boolean storeDoubleAsFloat;
+
+  @JsonCreator
+  public DoubleAnyAggregatorFactory(
+      @JsonProperty("name") String name,
+      @JsonProperty("fieldName") final String fieldName
+  )
+  {
+    Preconditions.checkNotNull(name, "Must have a valid, non-null aggregator name");
+    Preconditions.checkNotNull(fieldName, "Must have a valid, non-null fieldName");
+    this.name = name;
+    this.fieldName = fieldName;
+    this.storeDoubleAsFloat = ColumnHolder.storeDoubleAsFloat();
+  }
+
+  @Override
+  protected ColumnValueSelector selector(ColumnSelectorFactory metricFactory)
+  {
+    return metricFactory.makeColumnValueSelector(fieldName);
+  }
+
+  @Override
+  protected Aggregator factorize(ColumnSelectorFactory metricFactory, ColumnValueSelector selector)
+  {
+    return new DoubleAnyAggregator(selector);
+  }
+
+  @Override
+  protected BufferAggregator factorizeBuffered(ColumnSelectorFactory metricFactory, ColumnValueSelector selector)
+  {
+    return new DoubleAnyBufferAggregator(selector);
+  }
+
+  @Override
+  public Comparator getComparator()
+  {
+    return DoubleAnyAggregatorFactory.VALUE_COMPARATOR;
+  }
+
+  @Override
+  @Nullable
+  public Object combine(@Nullable Object lhs, @Nullable Object rhs)
+  {
+    if (lhs != null) {
+      return lhs;
+    } else {
+      return rhs;
+    }
+  }
+
+  @Override
+  public AggregatorFactory getCombiningFactory()
+  {
+    return new DoubleAnyAggregatorFactory(name, name);
+  }
+
+  @Override
+  public List<AggregatorFactory> getRequiredColumns()
+  {
+    return Collections.singletonList(new DoubleAnyAggregatorFactory(fieldName, fieldName));
+  }
+
+  @Override
+  public Object deserialize(Object object)
+  {
+    return object;
+  }
+
+  @Override
+  @Nullable
+  public Object finalizeComputation(@Nullable Object object)
+  {
+    return object;
+  }
+
+  @Override
+  @JsonProperty
+  public String getName()
+  {
+    return name;
+  }
+
+  @JsonProperty
+  public String getFieldName()
+  {
+    return fieldName;
+  }
+
+  @Override
+  public List<String> requiredFields()
+  {
+    return Arrays.asList(fieldName);
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    byte[] fieldNameBytes = StringUtils.toUtf8(fieldName);
+
+    return ByteBuffer.allocate(1 + fieldNameBytes.length)
+                     .put(AggregatorUtil.DOUBLE_ANY_CACHE_TYPE_ID)
+                     .put(fieldNameBytes)
+                     .array();
+  }
+
+  @Override
+  public String getTypeName()
+  {
+    if (storeDoubleAsFloat) {
+      return "float";
+    }
+    return "double";
+  }
+
+  @Override
+  public int getMaxIntermediateSize()
+  {
+    return Double.BYTES;
```

Comment:
not sure if this makes a difference or not, but won't the max size be `Float.BYTES` if storeDoubleAsFloat is true? 

I see the pattern you used is the same as what's used in `DoubleFirstAggregatorFactory` - probably a bug in both?

Reply:
Actually gonna remove storeDoubleAsFloat